pelican/plugins/sitemap/sitemap.py

```diff
@@ -55,235 +58,104 @@ def format_date(date):
-                "always",
-                "hourly",
-                "daily",
-                "weekly",
-                "monthly",
-                "yearly",
-                "never",
+        elif fmt == "xml":
+            lastmod = (
+                getattr(obj, "modified", None)
+                or getattr(obj, "date", None)
+                or datetime.combine(datetime.now(), datetime.min.time())
             )
-
-            if isinstance(pris, dict):
-                # We use items for Py3k compat. .iteritems() otherwise
-                for k, v in pris.items():
-                    if k in valid_keys and not isinstance(v, (int, float)):
-                        default = self.priorities[k]
-                        warning("sitemap plugin: priorities must be numbers")
-                        warning(
-                            "sitemap plugin: setting SITEMAP['priorities']"
-                            "['{0}'] on {1}".format(k, default)
-                        )
-                        pris[k] = default
-                self.priorities.update(pris)
-            elif pris is not None:
-                warning("sitemap plugin: SITEMAP['priorities'] must be a dict")
-                warning("sitemap plugin: using the default values")
-
-            if isinstance(chfreqs, dict):
-                # .items() for py3k compat.
-                for k, v in chfreqs.items():
-                    if k in valid_keys and v not in valid_chfreqs:
-                        default = self.changefreqs[k]
-                        warning("sitemap plugin: invalid changefreq `{0}'".format(v))
-                        warning(
-                            "sitemap plugin: setting SITEMAP['changefreqs']"
-                            "['{0}'] on '{1}'".format(k, default)
-                        )
-                        chfreqs[k] = default
-                self.changefreqs.update(chfreqs)
-            elif chfreqs is not None:
-                warning("sitemap plugin: SITEMAP['changefreqs'] must be a dict")
-                warning("sitemap plugin: using the default values")
-
-    def write_url(self, page, fd):  # NOQA C901
-
-        if getattr(page, "status", "published") != "published":
-            return
-
-        if getattr(page, "private", "False") == "True":
-            return
-
-        # We can disable categories/authors/etc by using False instead of ''
-        if not page.save_as:
-            return
-
-        page_path = os.path.join(self.output_path, page.save_as)
-        if not os.path.exists(page_path):
-            return
-
-        lastdate = getattr(page, "date", self.now)
-        try:
-            lastdate = self.get_date_modified(page, lastdate)
-        except ValueError:
-            warning(
-                "sitemap plugin: " + page.save_as + " has invalid modification date,"
+            content_type = {contents.Article: "articles", contents.Page: "pages"}.get(
+                type(obj), "indexes"
```

Comment:
Would be a lot more readable with regular `if` statements:
```
if isinstance(...):
    ...
elif isintance(...):
    ...
else:
    ...
```

Reply:
I now maybe used a ternary operator? I like how it conveys a single value expression and _not_ control flow.